### PR TITLE
[Keymap] Neo2 for ErgoDox on macOS US QWERTY / ABC Extended keymap

### PR DESCRIPTION
**Pull request by mjonuschat**
## Description

This is a Neo2 inspired layout that is meant to be fully usable on
macOS when used with the default US QWERTY/ABC Extended keymap.

Neo2 layers 1-4 have been almost fully implemented in hardware.
Layers 5 and 6 (greek and mathematical symbols) have been left
out for now as most of them aren't available on the default
keymaps.

Layer toggling for layer 3 on the right hand side utilizes a
tap-toggle approach that is a combination of MO & LT macros.
This is required to allow sending Y when tapped, @ when tapped
while the SHIFT modifier is active and support momentarily
toggling the layer while the key is held.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: layouts/community/ergodox/osx_neo2/layers.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#define NEO_1   0      // layer_0
```
Comment (noroadsleft) [inline layouts/community/ergodox/osx_neo2/layers.h]: Turning these defines into an `enum` will make these definitions easier to manage.

```c
enum layers {
    NEO_1,      // layer_0
    NEO_3,      // layer_1
    NEO_4,      // layer_2
    NEO_5,      // layer_3
    NEO_6,      // layer_4
    US_1,       // layer_5
    FKEYS       // layer_6
};
```

This is programmatically identical to your definitions, but doesn't require any manual defining.

File: layouts/community/ergodox/osx_neo2/README.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
+# Neo 2 for ErgoDox on QWERTY
```
[the remark this answers is not in the document]
Comment (mjonuschat) [reply layouts/community/ergodox/osx_neo2/README.md]: Done

File: layouts/community/ergodox/osx_neo2/keymap.c [hunk 3]
```diff
@@ -0,0 +1,708 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
[the remark this answers is not in the document]
Comment (mjonuschat) [reply layouts/community/ergodox/osx_neo2/keymap.c]: Done

File: layouts/community/ergodox/osx_neo2/keymap.c [hunk 4]
```diff
@@ -0,0 +1,708 @@
...
+// Timer to detect tap/hold on NEO_RMOD3 key
+static uint16_t neo3_timer;
+// State bitmap to track which key(s) enabled NEO_3 layer
+static uint8_t neo3_state = 0;
+// State bitmap to track key combo for CAPSLOCK
+static uint8_t capslock_state = 0;
+
+// bitmasks for modifier keys
+#define MODS_NONE   0
+#define MODS_SHIFT  (MOD_BIT(KC_LSHIFT)|MOD_BIT(KC_RSHIFT))
```
[the remark this answers is not in the document]
Comment (mjonuschat) [reply layouts/community/ergodox/osx_neo2/keymap.c]: Done

File: layouts/community/ergodox/osx_neo2/keymap.c [hunk 5]
```diff
@@ -0,0 +1,708 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
+#include "debug.h"
```
[the remark this answers is not in the document]
Comment (mjonuschat) [reply layouts/community/ergodox/osx_neo2/keymap.c]: Done

File: layouts/community/ergodox/osx_neo2/layers.h (hunk 1 above)
Comment (mjonuschat) [reply layouts/community/ergodox/osx_neo2/layers.h]: Done

**Pull request merged**